scripts/build/termux_download_deb_pac.sh

```diff
@@ -1,126 +1,79 @@
-				read -d "\n" PKG_PATH PKG_HASH <<<$(./scripts/get_hash_from_file.py "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH" $PACKAGE $VERSION)
-			elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-				if [ "$TERMUX_WITHOUT_DEPVERSION_BINDING" = "true" ] || [ $(jq -r '."'$PACKAGE'"."VERSION"' "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH") = "${VERSION_PACMAN}" ]; then
-					PKG_HASH=$(jq -r '."'$PACKAGE'"."SHA256SUM"' "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH")
-					PKG_PATH=$(jq -r '."'$PACKAGE'"."FILENAME"' "${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/$PACKAGE_FILE_PATH")
-					PKG_PATH="${PACKAGE_ARCH}/${PKG_PATH}"
-				fi
-			fi
-			if [ -n "$PKG_HASH" ] && [ "$PKG_HASH" != "null" ]; then
-				if [ ! "$TERMUX_QUIET_BUILD" = true ]; then
-					if [ "$TERMUX_REPO_PKG_FORMAT" = "debian" ]; then
-						echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx-1]}/dists/${TERMUX_REPO_DISTRIBUTION[$idx-1]}"
-					elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-						echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx-1]}"
-					fi
-				fi
-				break
-			fi
+		elif [[ -f "$TERMUX_COMMON_CACHEDIR-$PACKAGE_ARCH/$PACKAGE_FILE_PATH" ]]; then
+			read -d "\n" PKG_PATH PKG_HASH <<< "$(termux_download_deb_pac_get_pkg_hash_and_version "$PACKAGE_ARCH")" || :
+			[[ -n "$PKG_HASH" && "$PKG_HASH" != "null" ]] && break || :
+		elif [[ -f "$TERMUX_COMMON_CACHEDIR-aarch64/$PACKAGE_FILE_PATH" ]]; then
+			# Packages file for $PACKAGE_ARCH do not exist. Could be an aptly mirror where the
+			# all arch is mixed into the other arches, check for package in aarch64 Packages instead.
+			read -d "\n" PKG_PATH PKG_HASH <<< "$(termux_download_deb_pac_get_pkg_hash_and_version aarch64)" || :
+			[[ -n "$PKG_HASH" && "$PKG_HASH" != "null" ]] && break || :
 		fi
 	done
 
-	if [ "$PKG_HASH" = "" ] || [ "$PKG_HASH" = "null" ]; then
+	if [[ "$PKG_HASH" = "" || "$PKG_HASH" = "null" ]]; then
 		return 1
 	fi
 
-	termux_download "${TERMUX_REPO_URL[${idx}-1]}/${PKG_PATH}" \
-				"${TERMUX_COMMON_CACHEDIR}-${PACKAGE_ARCH}/${PKG_FILE}" \
+	termux_download "${TERMUX_REPO_URL[${idx}]}/$PKG_PATH" \
+				"$TERMUX_COMMON_CACHEDIR-$PACKAGE_ARCH/$PKG_FILE" \
 				"$PKG_HASH"
 }
 
+termux_download_deb_pac_get_pkg_hash_and_version() {
+	local arch="$1" pkg_file_path="$TERMUX_COMMON_CACHEDIR-$arch/$PACKAGE_FILE_PATH"
+	if $debian; then
+		read -d "\n" PKG_PATH PKG_HASH <<< "$(./scripts/get_hash_from_file.py "$pkg_file_path" $PACKAGE $VERSION)" || :
+	elif $pacman && [[ "$TERMUX_WITHOUT_DEPVERSION_BINDING" = "true" ]]; then
+		read -d "\n" PKG_PATH PKG_HASH <<< "$(jq -r '.["'"$PACKAGE"'"] | select(.VERSION=="'"$VERSION_PACMAN"'") | [.SHA256SUM, ("'"$arch"'"+"/"+.FILENAME)] | @tsv' "$pkg_file_path")" || :
+	fi
+	if [[ -n "$PKG_HASH" && "$PKG_HASH" != "null" && "$TERMUX_QUIET_BUILD" != "true" ]]; then
+		echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx]}$($debian && echo "/dists/${TERMUX_REPO_DISTRIBUTION[$idx]}" || :)" >&2
```

Comment:
No using a variable as a command please.
```suggestion
		echo "Found $PACKAGE in ${TERMUX_REPO_URL[$idx]}$([[ "$debian" == 'true' ]] && echo "/dists/${TERMUX_REPO_DISTRIBUTION[$idx]}" || :)" >&2
```

Reply:
Done.